first-task/src/LengthConverte.js

```diff
@@ -0,0 +1,99 @@
+const { type } = require('./utils');
+
+const resultConvert = (value, measurement, obj) => {
+  if (
+    type(value) === 'number'
+    && value >= 0
+    && type(measurement) === 'string'
+    && Object.keys(obj).some(a => a === measurement)
+  ) {
+    if (value === 0) return 0;
+    if (measurement === 'sea miles') return value * obj[measurement];
+
+    return value / obj[measurement];
+  }
+
+  throw new Error('error');
+};
+
+class LengthConverter {
+  static convertToMiles(value, measurement) {
+    const CONVERT_TO_MILES = {
```

Comment:
Why is it so complicated? It could be done much easier via `mediator`. As result you have only 2 tables of units.

For example you can have coefficients values for converting to meters, and coefficients for converting from meters to anything.

Example when we use SeaMiles as mediator:
```
const measurmentsCoefToSeaMilesMap = new Map([
    [measurements.CM, 5.3996e-6],
    [measurements.M, 0.000539957],
    [measurements.Miles, 0.868976],
    [measurements.valueInSM, 1],
    [measurements.In, 1.3715e-5],
    [measurements.Ft, 0.000164579]
]);
const SeaMilesCoefToAnyMap = new Map([
    [measurements.CM, 185200],
    [measurements.M, 1852],
    [measurements.Miles, 1.15],
    [measurements.valueInSM, 1],
    [measurements.In, 72913.4],
    [measurements.Ft, 6076.12]
]);
```

Reply:
Then we can convert for example from any to sea miles, and from sea miles then to anything else. I suggest to use meters, there will more accuracy during conversion.